### ISSUE TITLE
Today's 1.29 Insiders Breaks Extensions

### ISSUE THREAD

**Issue by Astrantia** [Microsoft/vscode#60168]
Logs:
```
[Extension Host] Activating extension `vscodevim.vim` failed:  Cannot read property 'prototype' of undefined (at c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\node\extensionHostProcess.js:295:9)
console.ts:134 
[Extension Host] Activating extension 'vscodevim.vim' failed: Cannot read property 'prototype' of undefined. (at Object.showMessage (c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\node\extensionHostProcess.js:623:628)) console.ts:134 
[Extension Host] Activating extension 'esbenp.prettier-vscode' failed: Super expression must either be null or a function. (at Object.showMessage (c:\Program Files\Microsoft VS Code\Insiders\resources\app\out\vs\workbench\node\extensionHostProcess.js:623:628))
console.ts:134 
[Extension Host] Activating extension `esbenp.prettier-vscode` failed:  Super expression must either be null or a function (at c:\Program Files\Microsoft VS Code Insiders\resources\app\out\vs\workbench\node\extensionHostProcess.js:295:9)
```

**Comment by Astrantia** [comment]
cc @bpasero @sandy081

**Comment by Astrantia** [comment]
Note that this didn't break all of my extensions, only some:
```
CoenraadS.bracket-pair-colorizer
esbenp.prettier-vscode
vscodevim.vim
```

**Comment by JCMais** [comment]
Same issue here, using Insiders
```
Version: 1.29.0-insider (user setup)
Commit: 1358e903716cf77711493770d63dde1ea178a73e
Date: 2018-10-08T05:16:11.119Z
Electron: 2.0.9
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64
```

**Comment by carlocardella** [comment]
same: 

Version: 1.29.0-insider (user setup)
Commit: 1358e903716cf77711493770d63dde1ea178a73e
Date: 2018-10-08T05:16:11.119Z
Electron: 2.0.9
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

**Comment by egamma** [comment]
Cannot reproduce in stable 1.28 and the comments above are all from insiders.

**Comment by suciuvlad** [comment]
+1

**Comment by sstoychev** [comment]
Same here, broke liximomo.sftp although I don't see any errors in the logs.

Version: 1.29.0-insider (user setup)
Commit: 1358e903716cf77711493770d63dde1ea178a73e
Date: 2018-10-08T05:16:11.119Z
Electron: 2.0.9
Chrome: 61.0.3163.100
Node.js: 8.9.3
V8: 6.1.534.41
Architecture: x64

**Comment by DonJayamanne** [comment]
Most probably a dup of https://github.com/Microsoft/vscode/issues/60190 (`global` variable is incorrect in npm modules)

**Comment by ArtemGovorov** [comment]
Looks like some loader/require hooks changes had broken node globals. 

Here is a simple way to reproduce the issue:

extension.js

```js
require('./initGlobals');
console.log(myGlobal); // <-- breaks, but used to work
```

initGlobals.js

```js
global.myGlobal = 'test';
```

**Comment by Astrantia** [comment]
@Tyriar @bpasero Can we get recoveries for Insiders within the same day when there's issues like this? I only use Insiders so I would appreciate more attention over here.

**Comment by BlackHoleFox** [comment]
Had this issue as well, it completely broke the Microsoft Python extension.

**Comment by jrieken** [comment]
This might be related to us now using cached data for extensions

**Comment by jrieken** [comment]
The is fixed in 1.29.0 `c1100eff93b35dc220ce63e2ed0d5ace54259795`
